src/js/time-manager.js

```diff
@@ -59,36 +68,54 @@ class TimeManager extends LitElement {
   }
 
   async setWorkedHours() {
+    // set up
     const storedTimes = await getTimePairs('time-pairs-only');
     const nowDecimal = human2decimalTime();
-    let newWorkedHours = 0
+    let newWorkedHours = 0;
+    let lastStopTime = 0;
+    let pauses = 0;
+    // for all time-interval pairs:
+    storedTimes.forEach(([startTime, stopTime], i) => {
+      // valid start time?
+      if (isNaN(startTime) || startTime > nowDecimal) {
+        // no, so skip this time interval
+        return;
+      }
+      // for first time interval only, ...
+      if (i === 0) {
+        // ... remember it as start-of-day
+        this.dayStart = startTime;
+        // ... and initialize pseudo last-stop-time
+        lastStopTime = startTime;
+      }
 
-    storedTimes.forEach(([startTime, stopTime]) => {
-      if(startTime) {
-        const stopTimeIsNaN = isNaN(stopTime) || stopTime === undefined;
+      // ensure valid stop time
+      stopTime = isNaN(stopTime)
```

Comment:
if no stopTime is given it could be undefined. So better check it this way: `const stopTimeIsNaN = isNaN(stopTime) || stopTime === undefined;`